frame/support/procedural/src/pallet/expand/error.rs

```diff
@@ -39,13 +50,36 @@ pub fn expand_error(def: &mut Def) -> proc_macro2::TokenStream {
 		)
 	);
 
-	let as_u8_matches = error.variants.iter().enumerate().map(
-		|(i, (variant, _))| quote::quote_spanned!(error.attr_span => Self::#variant => #i as u8,),
-	);
-
-	let as_str_matches = error.variants.iter().map(|(variant, _)| {
+	let as_u8_matches =
```

Comment:
we must remove as_u8 as an error is no longer represented as an u8. Instead its representation is its encoding now.

That also mean we need to change the type DispatchError. The variant Module, must have a field error as `Vec<u8>` as it is now the encoding of the error and no longer a simple u8.
This broader change will need to be notified to external tools.

Reply:
Wow ok, this is a bigger change than I thought. Introducing a `Vec<u8>` there means that `DispatchError` can no longer derive `Copy`, and it also means that this type is no longer bounded in size. If this is the proper way that we should implement things, I think we should at least set a sensible bound on this.

Reply:
yes maybe we can have error being an an array `[u8; static_depth]`.
But then any depth beyond the static_depth will be ignored, that can be fine, maybe we can also check at runtime or compiletime.